Cabal/Distribution/Types/PackageId.hs

```diff
@@ -40,15 +37,61 @@ instance Pretty PackageIdentifier where
     | v == nullVersion = pretty n -- if no version, don't show version.
     | otherwise        = pretty n <<>> Disp.char '-' <<>> pretty v
 
+-- |
+--
+-- >>> simpleParse "foo-bar-0" :: Maybe PackageIdentifier
+-- Just (PackageIdentifier {pkgName = PackageName "foo-bar", pkgVersion = mkVersion [0]})
+--
+-- >>> simpleParse "foo-bar" :: Maybe PackageIdentifier
+-- Just (PackageIdentifier {pkgName = PackageName "foo-bar", pkgVersion = mkVersion []})
+--
+-- /Note:/ Broken (too lenient, doesn't require full consumption)
```

Comment:
Issue for this? Or do we not care because this instance is going away? In which case, the comment should probably say so.

Reply:
Implied by #5579 . No `ReadP` => No `Text`